libcaf_net/caf/net/http/request_header.test.cpp

```diff
@@ -0,0 +1,71 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/net/http/request_header.hpp"
+
+#include "caf/test/caf_test_main.hpp"
+#include "caf/test/test.hpp"
+
+using namespace caf;
+using namespace std::literals;
+
+TEST("parsing a http request") {
+  net::http::request_header hdr;
+  hdr.parse("GET /foo/bar?user=foo&pw=bar#baz HTTP/1.1\r\n"
+            "Host: localhost:8090\r\n"
+            "User-Agent: AwesomeLib/1.0\r\n"
+            "Accept-Encoding: gzip\r\n"
+            "Number: 150\r\n\r\n");
+  require(hdr.valid());
+  SECTION("check value accessors") {
+    check_eq(hdr.method(), net::http::method::get);
+    check_eq(hdr.version(), "HTTP/1.1");
+    check_eq(hdr.path(), "/foo/bar");
+    check_eq(hdr.query().at("user"), "foo");
+    check_eq(hdr.query().at("pw"), "bar");
+    check_eq(hdr.fragment(), "baz");
+    check_eq(hdr.num_fields(), 4ul);
+    check_eq(hdr.field("Host"), "localhost:8090");
+    check_eq(hdr.field("User-Agent"), "AwesomeLib/1.0");
+    check_eq(hdr.field("Accept-Encoding"), "gzip");
+  }
+  SECTION("fields access is case insensitive") {
+    check_eq(hdr.field("HOST"), "localhost:8090");
+    check_eq(hdr.field("USER-agent"), "AwesomeLib/1.0");
+    check_eq(hdr.field("accept-ENCODING"), "gzip");
+  }
+  SECTION("non existing field returns an empty view") {
+    check_eq(hdr.field("Foo"), "");
+  }
+  SECTION("field access by position") {
+    check_eq(hdr.field_at(0), std::pair("Host"sv, "localhost:8090"sv));
```

Comment:
```suggestion
    check_eq(hdr.field_at(0), std::pair{"Host"sv, "localhost:8090"sv});
```

(same for the others)